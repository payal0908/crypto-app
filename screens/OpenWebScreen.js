import React, {useState} from 'react';
import { WebView } from 'react-native-webview';
import {View, Text, StyleSheet} from 'react-native';
import * as Progress from 'react-native-progress';

const OpenWebScreen = ({navigation, route}) => {

  const [progress, setProgress] = useState(0);
  const [isLoaded, setLoaded] = useState(false);

  const url = route.params.url;

  return (
    <>
    {
      !isLoaded ?
      <Progress.Bar progress={progress} width={null} borderWidth={0} borderRadius={0} color='coral' />
      :
      null
    }
      <WebView
        source={{ uri: url }}
        onError={(event) => alert(`${event.nativeEvent.description}`)}
        onLoadEnd={() => setLoaded(true)}
        onLoadProgress = {({nativeEvent}) => setProgress(nativeEvent.progress)}
      />
    </>
  )
}

export default OpenWebScreen;

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#222',
      alignItems: 'center',
      justifyContent: 'center'
    },

});
