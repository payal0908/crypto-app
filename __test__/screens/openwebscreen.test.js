import React from 'react';
import {create} from 'react-test-renderer';

import OpenWebScreen from '../../screens/OpenWebScreen';



test('snapshot', () => {
    const mockedParams = {
        route: { params: { url: 'url' } },
        navigation: ''
      };
    const tree = create(<OpenWebScreen {...mockedParams}/>)
    expect(tree).toMatchSnapshot()
});
