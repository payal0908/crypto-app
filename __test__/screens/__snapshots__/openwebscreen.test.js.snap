// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <View
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    indeterminate={false}
    indeterminateAnimationDuration={1000}
    onLayout={[Function]}
    progress={0}
    style={
      Array [
        Object {
          "backgroundColor": undefined,
          "borderColor": "coral",
          "borderRadius": 0,
          "borderWidth": 0,
          "overflow": "hidden",
          "width": null,
        },
        undefined,
      ]
    }
    useNativeDriver={false}
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "coral",
          "height": 6,
          "transform": Array [
            Object {
              "translateX": -0,
            },
            Object {
              "translateX": -0,
            },
            Object {
              "scaleX": 0.0001,
            },
          ],
        }
      }
    />
  </View>,
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onError={[Function]}
      onHttpError={[Function]}
      onLoadEnd={[Function]}
      onLoadProgress={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "whatever-id",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      textInteractionEnabled={true}
      useSharedProcessPool={true}
    />
  </View>,
]
`;
